### DISCUSSION TITLE
Merge branch 'main' of repo into 'main'

### DISCUSSION THREAD

**Discussion by bertmelis** [community/community#33361] [Pull Requests] [answered]
I've got a commit that I don't know he origin of.
While merging a PR I got an error message the head changed. But a few moments later the error was gone and merging went without issue. Later I did another commit and created a release. Meanwhile I also updated the gh-pages branch.

Now the latest commit -which I did not knowingly made- has the mysterious title "[Merge branch 'main' of github.com:bertmelis/espMqttClient into main](https://github.com/bertmelis/espMqttClient/commits/main)". Furthermore, the commit has an "unverified" signature.

Details of the commit are the same as the PR with the error.

Does anybody know what went wrong? How can I prevent this in the future?

**Comment by bertmelis** [comment]
Local git log shows this:

```
commit 62e522737082d6fe0f628ba3ae25979d09df7296 (HEAD -> main, tag: v1.3.0, origin/main, origin/HEAD)
Merge: 0af3f5a 9ab7149
Author: Bert Melis <bertmelis@users.noreply.github.com>
Date:   Sat Sep 17 11:39:14 2022 +0200

    Merge branch 'main' of github.com:bertmelis/espMqttClient into main

commit 0af3f5af4e42722b223f9469a9c001da731b076c
Author: Bert Melis <bertmelis@users.noreply.github.com>
Date:   Sat Sep 17 11:39:10 2022 +0200

    bump version

commit 9ab7149ca4bd2fad4edd6b727fb2741efae2d7d3
Author: Bert Melis <bertmelis@users.noreply.github.com>
Date:   Sat Sep 17 11:29:41 2022 +0200

    Linux compatibility

    - run on Linux*
    - add `void mqttClient.disconnected()` to the API
    - split rx and tx buffer size
```

**Comment by airtower-luna** [comment]
That last commit looks like a regular merge commit as I'd expect if the two commits before had been developed on separate branches (branches of the same name in different copies of the repository count) which were then merged. Why that might have happened is hard to say without knowing the exact steps you did, their order, and so on. :sweat_smile:

**Reply by bertmelis** [reply]
1. (squash) merge from branch
2. version update directly on main

I might not have synced between one and two before comitting the version update. So my local "main" was not up-to-date with "main" from github.

**Reply by airtower-luna** [reply, marked as answer]
That would explain it, if you then did a pull before tagging. AFAIK it's the default that `git pull` will merge if needed. If you did the merge on GitHub, committed the version update locally, and *then* pulled the result would be what you see. :slightly_smiling_face: 

I've set [`pull.ff = only`](https://git-scm.com/docs/git-config#Documentation/git-config.txt-pullff) in my Git config to prevent `git pull` from merging automatically (of course you can still merge manually if you want), maybe that's interesting for you.
